types/indexer.go

```diff
@@ -0,0 +1,292 @@
+package types
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"fmt"
+	"github.com/jordanorelli/lexnum"
+	"github.com/pkg/errors"
+	"github.com/tendermint/tendermint/libs/pubsub/query"
+	"github.com/tendermint/tendermint/state/txindex"
+	"github.com/tendermint/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+	"math"
+)
+
+var (
+	_ txindex.TxIndexer = &TransactionIndexer{}
+	e                   = lexnum.NewEncoder('=', '-')
```

Comment:
I think `e` is probably not a good way to name this variable since its global.

Reply:
Agreed, will update